lib/manager/cargo/extract.ts

```diff
@@ -63,13 +83,50 @@ function extractFromSection(
 
 export function extractPackageFile(
   content: string,
-  fileName: string
+  fileName: string,
+  config?: ExtractConfig
 ): PackageFile | null {
   logger.trace(`cargo.extractPackageFile(${fileName})`);
-  let parsedContent: CargoConfig;
+
+  let cargoConfig: CargoConfig | undefined;
+  if (config?.localDir) {
+    let configPath = join(config.localDir, '.cargo', 'config.toml');
```

Comment:
Reading the `.cargo/config.toml` (or the legacy `.cargo/config` - same format, new name) here is probably not ideal, I understand `extractPackageFile` is only supposed to parse an already-read file, but I don't think the `ManagerApi` allows doing this any other way?

I'm open to suggestions on this one.